src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -2599,7 +2624,7 @@ void EmitNegativeLookaroundAssertion(RegexNode node)
                     isInLoop = rm.Analysis.IsInLoop(node);
                     if (isInLoop)
                     {
-                        EmitStackPush("base.Crawlpos()");
+                        EmitStackPush(stackCookie, "base.Crawlpos()");
```

Comment:
Sort of unrelated to your PR, but not sure I'm following why we are not pushing to the stack in Positive look arounds too

Reply:
https://github.com/dotnet/runtime/pull/97463

Captures for positive lookarounds persist to outside of the lookaround, whereas captures for negative lookarounds are always uncaptured to not persist after the lookaround.

Reply:
ah -- I either wasn't aware of that, or have forgotten it 😅😝

Reply:
I wasn't aware, which is why I had to fix it last month :)